### PR TITLE
Align `DF.pivot_longer/3` API with dplyr

### LINKED ISSUE #241: Align pivot_longer with dplyr
In dplyr, the `cols` argument to `pivot_longer` is actually the columns we would like to pivot. I find that API a bit more intuitive. Should we change our implementation to align with theirs? :)

### PR DESCRIPTION
**Pull request by philss**
Reference:
https://tidyr.tidyverse.org/reference/pivot_longer.html

Closes #241

### CODE REVIEW

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -1756,85 +1761,93 @@ defmodule Explorer.DataFrame do
...
-      |> Map.take(columns)
+      |> Map.take(keep_columns)
       |> Map.put(names_to, :string)
       |> Map.put(values_to, values_dtype)
 
-    out_df = %{df | names: columns ++ [names_to, values_to], dtypes: new_dtypes}
+    out_df = %{df | names: keep_columns ++ [names_to, values_to], dtypes: new_dtypes}
 
-    Shared.apply_impl(df, :pivot_longer, [out_df, value_columns])
+    Shared.apply_impl(df, :pivot_longer, [out_df, columns_to_pivot])
```
Comment (josevalim) [inline lib/explorer/data_frame.ex:1850]: Let's pass `keep`, `names_to`, and `values_to` to the backend, so it doesn't have to compute it. WDYT? Or at least `names_to` and `values_to`.

**Comment by cigrainger** [comment]
@jsonbecker Can I please get your input on the API and naming here? R has been through so many iterations of this I'm not even sure what the pros and cons are anymore.

**Comment by jsonbecker** [comment]
@cigrainger I think having `names_to` and `values_to` are solid as naming conventions. I also like `columns_to_pivot`.

`keep` is interesting. I think of `keep` as defining ids or keys, conceptually, but I think `keep` would be well understood.

File: lib/explorer/data_frame.ex (hunk 1 above)
Comment (philss) [reply lib/explorer/data_frame.ex:1850]: yeah, makes sense!

**Comment by philss** [comment]
@jsonbecker cool, I liked the suggestions! Thanks!

**Review by cigrainger — APPROVED** [review]
Looks good to me! <3 Ship it!

**Pull request merged**